pkg/apis/core/validation/shoot.go

```diff
@@ -878,6 +878,10 @@ func ValidateClusterAutoscaler(autoScaler core.ClusterAutoscaler, fldPath *field
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("expander"), *expander, availableClusterAutoscalerExpanderModes.List()))
 	}
 
+	if ignoreTaints := autoScaler.IgnoreTaints; ignoreTaints != nil && !isIgnoreTaintsUnique(ignoreTaints) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("ignoreTaints"), ignoreTaints, "must have unique keys"))
+	}
```

Comment:
Are you sure? Because, over here the taints are being listed to be ignored & this list is not per worker pool as such; it is for all the nodes of the cluster as a whole. I am not quite sure if similar worker taint key validation would be relevant here.

Reply:
I rather suggested that you can check how taint key is validated for worker taints and we could reuse the same validation handling for `ignoreTaints` (that represents a slice of taint keys). Do you have any concerns with this approach?

Reply:
The datatype of taints in the `validateTaints` for worker is of type `[]corev1.Taints` where as that if `ignoreTaints` is `[]string`. So, I am not sure how this can be reused?

Reply:
Correct, but this suggestion was rather to check what is the validation performed for the taint key in `validateTaints` and to do the same validation for `ignoreTaints`.

You can see that the taint key is validated with `metav1validation.ValidateLabelName`.

https://github.com/gardener/gardener/blob/ce9d741798eac2df8c470190ab483aa4c5818ebf/pkg/apis/core/validation/shoot.go#L1301-L1302